### PR TITLE
[Nanopb] 0.4.7

### PR DESCRIPTION
**Pull request by eharshosh**
**Describe the pull request**

- #### What does your PR fix?
Upgrades nanopb port to 0.4.7

- #### Which triplets are supported/not supported? Have you updated the [CI baseline]
no changes as far as I can tell

- #### Does your PR follow the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)?
  I believe so

- #### If you have added/updated a port: Have you run `./vcpkg x-add-version --all` and committed the result?
  Yes

**If you are still working on the PR, open it as a Draft: https://github.blog/2019-02-14-introducing-draft-pull-requests/**

### CODE REVIEW

**Comment by eharshosh** [comment]
@microsoft-github-policy-service agree [company="SolarGik"]

**Comment by eharshosh** [comment]
@microsoft-github-policy-service agree company="SolarGik"

File: ports/nanopb/vcpkg.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "nanopb",
-  "version-semver": "0.4.5",
+  "version-semver": "0.4.7",
   "description": "A small code-size Protocol Buffers implementation in ANSI C.",
   "homepage": "https://jpa.kapsi.fi/nanopb/",
```
Comment (MonicaLiu0311) [inline ports/nanopb/vcpkg.json:5]:
```suggestion
  "homepage": "https://jpa.kapsi.fi/nanopb/",
  "license": "Zlib",
```

A `license` is required.

**Comment by MonicaLiu0311** [comment]
The file contains at least one deprecated function, we recommend the new function if you find it convenient:
`vcpkg_configure_cmake` -> `vcpkg_cmake_configure` (Please remove the option PREFER_NINJA) (from port `vcpkg-cmake`)
`vcpkg_install_cmake` -> `vcpkg_cmake_install` (from port `vcpkg-cmake`)
`vcpkg_fixup_cmake_targets` -> `vcpkg_cmake_config_fixup` (from port `vcpkg-cmake-config`)

In the ports that use the new function, you have to add the corresponding dependencies:
```
{
  "name": "vcpkg-cmake",
  "host": true
},
{
  "name": "vcpkg-cmake-config",
  "host": true
}

**Comment by MonicaLiu0311** [comment]
```
# Handle copyright
configure_file(${SOURCE_PATH}/LICENSE.txt ${CURRENT_PACKAGES_DIR}/share/${PORT}/copyright COPYONLY)
```
One of the following two methods is more recommended:
* file(INSTALL "${SOURCE_PATH}/LICENSE.txt" DESTINATION "${CURRENT_PACKAGES_DIR}/share/${PORT}" RENAME copyright)
* vcpkg_install_copyright(FILE_LIST "${SOURCE_PATH}/LICENSE.txt")

**Comment by MonicaLiu0311** [comment]
All absolute paths in `portfile.cmake` must be enclosed in double quotes.

**Comment by MonicaLiu0311** [comment]
Feature `generator` are tested successfully in the following triplet:
x86-windows
x64-windows
x64-windows-static

**Review by MonicaLiu0311 — APPROVED** [review]

**Comment by BillyONeal** [comment]
Thanks!

**Comment by YusufRamlawi** [comment]
Thanks Eitan

**Pull request merged**